### DISCUSSION TITLE
Confusion with Script inside _document file and no-script-in-document-page ESLint rule

### DISCUSSION THREAD

**Discussion by emiliosheinz** [vercel/next.js#37413] [Help] [answered]
I'm trying to add a Script with `strategy="beforeInteractive"` to my application.  Reading [the docs](https://nextjs.org/docs/basic-features/script#beforeinteractive) I saw that this strategy only works inside `_document` file. After adding the Script to the `_document` file ESLint started to complain about using scripts inside this file, this is caused by [this rule](https://nextjs.org/docs/messages/no-script-in-document-page). 

Does anyone know which is the correct way of using scripts with `strategy="beforeInteractive"`? The NextJS docs seem to be ambiguous about it.

Some environment info:
```
NextJS: 12.1.6
eslint-config-next: 12.1.5
```

**Comment by icyJoseph** [comment, marked as answer]
Hi,

The NextJS family of packages ought to be at the same version, otherwise this kind of thing happens. Please update to `eslint-config-next` **12.1.6**. The same applies if you have `@next/bundle-analyzer`.

#### A bug 🐛 

In version **12.1.6**, AFAIK, the `beforeInteractive` scripts in `_document`, are not really working... https://github.com/vercel/next.js/pull/37000, it has been fixed, but the fix was not as desired... https://github.com/vercel/next.js/issues/37225#issuecomment-1142390675, there has been a bit of a back and forth... https://github.com/vercel/next.js/pull/37084, so for now, I'd say under 12.1.6, please use a regular script, and as soon as a new version comes out switch back to `next/script` with `beforeInteractive` set.

**Comment by emiliosheinz** [comment]
@icyJoseph Thank you for the quick response. What would be the equivalent of `next/script` with  `beforeInteractive` inside `_document`?

**Reply by icyJoseph** [reply]
Right now, a plain `<script></script>` with its `src` set to what you need. When they release the fix, just upgrade it to `<Script />` and put the `beforeInteractive` strategy on it.

```tsx
<script
  type="text/javascript"
  src="https://unpkg.com/lodash@4.17.20/lodash.js"
></script>
```

**Reply by emiliosheinz** [reply]
Nice! Is there any specific place where I should place it? Now I have it inside `_document` file inside a `Head` tag, is this good to go?

**Reply by icyJoseph** [reply]
Wherever it works for you :) - I think inside Head is preferred though.

**Reply by emiliosheinz** [reply]
Ok, thank you for the help 🤙🏼
